eds4jinja2/adapters/tabular_utils.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/python3
+
+# tabular_utils.py
+# Date:  17/08/2020
+# Author: Eugeniu Costetchi
+# Email: costezki.eugen@gmail.com 
+
+""" a set of helper functions to easily polish the tabular data (i.e Pandas DataFrame) """
+import re
+from typing import List, Dict
+
+import pandas as pd
+import numpy as np
+import logging
+
+from pandas.core.dtypes.common import is_numeric_dtype
+
+logger = logging.getLogger(__name__)
+
+
+def invert_dict(mapping_dict: Dict, reduce_values: bool = True):
+    """
+        Invert the dictionary by swapping keys and values. In case the values are unique then the inverted dict will be
+        of the same size as the initial one. Otherwise it will be shrunk to the unique values and the keys will be
+        cumulated in a list.
+
+        The list can be reduced to single item by setting reduce_values=True.
+
+        >>> d = {"a":1, "b":2, c:1}
+        >>> reduced_d = invert_dict(d)
+        {1: 'a', 2: 'b'}
+
+        >>> unreduced_d = invert_dict(d, False)
+        {1: ['a', 'c'], 2: ['b']}
+
+        :param reduce_values: If reduce_values is true then the values are single items otherwise
+                                the values are list of possibly multiple items.
+        :type mapping_dict: a dictionary to be inverted
+    """
+    inv_map = {}
+    for k, v in mapping_dict.items():
+        inv_map[v] = inv_map.get(v, [])
+        inv_map[v].append(k)
+    if reduce_values:
+        return {k: sorted(v, key=len)[0] for k, v in inv_map.items()}
+    return inv_map
+
+
+def replace_strings_in_tabular(data_frame: pd.DataFrame, target_columns: List[str] = [], value_mapping_dict: Dict = {},
```

Comment:
you should not use [mutables as default values](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)

instead of:
```python
def replace_strings_in_tabular(data_frame: pd.DataFrame, target_columns: List[str] = [], value_mapping_dict: Dict = {}, mark_touched_rows: bool = False):
```
do:
```python
def replace_strings_in_tabular(data_frame: pd.DataFrame, target_columns: List[str] = None, value_mapping_dict: Dict = None, mark_touched_rows: bool = False):
    if target_columns is None:
        target_columns = []
    if value_mapping_dict is None:
        value_mapping_dict = {}
```

Reply:
Thank you, I did not know what the implications are. 
In this case, they are still safe because no modification is performed on them, however, I implemented your suggestion.